### ISSUE TITLE
[BUG] Chia 1.2.5 freezes RPi4B 4GB during full node sync

### ISSUE THREAD

**Issue by Zeesy** [Chia-Network/chia-blockchain#8344]
**Describe the bug**

After starting Chia 1.2.5 on an RPi running Ubuntu, the computer freezes before sync advances. This occurs with GUI and CLI.

**To Reproduce**

1. Start Chia either through GUI or CLI
2. full node will begin to sync (yellow text)
3. Computer freezes before sync advances past initial difficulty 

**Expected behavior**

Expecting that full node will take some time to sync, but will eventually sync without freezing computer as had happened in previous Chia versions.

**Desktop**

- OS: Ubuntu Linux 21.04
- OS Version/Flavor: Hirstute
- CPU: RPi 4B 4GB RAM Broadcom BCM2711, Quad core Cortex-A72 (ARM v8) 64-bit SoC @ 1.5GHz

**Additional context**

Is the RPi 4B no longer enought to run a full node? See issue #8341 for GUI error logs

**Comment by jack60612** [comment]
There are memory leaks rn so it should work but you might have to try the main branch and build from source

**Comment by Zeesy** [comment]
Tried building from source but still got `npm ERR! code ENOENT` and associated errors after running `npm run electron &`. The process stalls before gui can start. Just tried a clean install based on the docs, Pi now freezing after language pack and electron installation for `sh install-gui.sh`.

**Comment by denisprovost** [comment]
I'm on Ubuntu 21.04 on pi4 8g, after resync the system use 3.8giga. 4 risk of being light

**Comment by Zeesy** [comment]
USB-C Switching power adapter, 100-240V~50/60Hz input, 5.1V = 3000mA output
Came as part of a PiShop kit.

Does this mean the challenges are becoming too complex to run on a Pi with less than 8g RAM?

**Comment by denisprovost** [comment]
I think this may be more related to the Linux distro used. Some do this on Pi4 4G with some distro like Arch it seems to me, ubuntu 21.04 is greedy.

In your case I would go for a problem of lack of power, test with a power adapter 5V 5A maybe.

**Comment by loppefaaret** [comment]
3A should be sufficient without overclocking, or USB devices drawing power(2.5 externals powered by host)
running the desktop and GUI does put alot more pressure on a pi, even a pi4 - and as mentioned above, there is a network issue currently affecting all release clients, that increases the memory usage of the node - i would recommend to revert to CLI usage until we atleast get a fix out

**Comment by emlowe** [comment]
I don't recommend running the GUI on a 4GB pi4 . It's better to run the services from the CLI. Also, it's very likely you have experienced the node memory issue (#8307)

**Comment by denisprovost** [comment]
@emlowe
I sended this issue ([#8307](https://github.com/Chia-Network/chia-blockchain/issues/8307))

On Pi4, this memory leak is not a problem during syncing, it only becomes noticeable after several hours.

**Comment by Zeesy** [comment]
> @emlowe
> 
> On Pi4, this memory leak is not a problem during syncing, it only becomes noticeable after several hours.

So I'm having a separate issue from the memory leak? Will try strictly CLI and see how that goes. @denisprovost , are you running raspbian or ubuntu?

**Comment by denisprovost** [comment]
I use Ubuntu 21.04 on PI 4 8giga, but with the server version of ubuntu. Your problem may simply be a lack of memory for Pi4 4giga for the chia gui process.

**Comment by Zeesy** [comment]
How long do you think until 8G is too little? It's a shame to lose out on these nice little machines.

**Comment by denisprovost** [comment]
I don't know but rpi will have little more ram available in the future. For the moment 8gigas is enough for the complete knot + haverster.

**Comment by emlowe** [comment]
> @emlowe
> I sended this issue ([#8307](https://github.com/Chia-Network/chia-blockchain/issues/8307)), I didn't mention that this happens during syncing.
> 
> On Pi4, this memory leak is not a problem during syncing, it only becomes noticeable after several hours. Making a shortcut to this problem for each problem encountered without an answer is not possible ;)

This is true, I had forgotten this - the memory leak can only happen *after* being fully synced. Note running the full node/farmer is supported on the pi4 4GB, however, typically we recommend only doing this using the CLI. Running the GUI does require a fair amount of resources. The pi4 also does work very well as a remote harvester

**Comment by emlowe** [comment]
Closing issue as there isn't anything actionable in the report. The pi4 4GB is supported as a node. If your node is not syncing we will need logs to determine why

**Comment by Zeesy** [comment]
Could you provide some instructions on submitting logs? The CLI is crashing on 1.2.7, same 4G rpi

**Comment by jack60612** [comment]
> Could you provide some instructions on submitting logs? The CLI is crashing on 1.2.7, same 4G rpi

@Zeesy  message jacknelson on keybase with your debug log

**Comment by jack60612** [comment]
I'm reviewing the logs now

**Comment by jack60612** [comment]
Logs are showing that the node dies without an error of any kind. 
I think this is an issue with running the gui on the rpi 4 as the 4gbs of ram is going to be eaten by ubuntu.
I would recommend installing [Ubuntu Server 20.04.3 64 bit](https://ubuntu.com/download/raspberry-pi)
Using ssh for the connection and installing chia from [source](https://github.com/Chia-Network/chia-blockchain/wiki/INSTALL#ubuntudebian) as you don't want to install the gui which will cause this all over again.

**Comment by isipisi281** [comment]
4 gig is plenty IF you have 4gb extra swap space. 
Take a 4GB usb stick. Start gparted and format it as swap. than right click and turn swap on.
i have 60 gig of swap on an nvme and overclockt the pi4 4gb to 1.9 ghz and its running fine.

![swap](https://user-images.githubusercontent.com/8292225/134599416-74ec50da-3f0b-4152-acb7-5eb168e7002d.PNG)

**Comment by chapas** [comment]
I'm currently with a RPi400, 4GB, currently 14 USB HDD drives, ~130TB, 5V 10A power supply.
In terms of memory, it becomes unstable while running a full node. With a Rpi4 8Gb might work, but not 4GB. Added 4Gb of zram/swapp it does not crash.
Plus I'm having issues with the network, which is failing 15sec. from 2 to 2 minutes, due to `iowait` I believe.
This is not a verdict, but, in my opinion, with the growth on DB, more I/O, mem. required, "this" will not work for much longer.

![image](https://user-images.githubusercontent.com/83612878/134655521-c17bb870-57b7-4c61-b6c9-b3c2be6c4ecb.png)

**Comment by isipisi281** [comment]
> I'm currently with a RPi400, 4GB, currently 14 USB HDD drives, ~130TB, 5V 10A power supply.
> In terms of memory, it becomes unstable while running a full node. With a Rpi4 8Gb might work, but not 4GB. Added 4Gb of zram/swapp it does not crash.
> Plus I'm having issues with the network, which is failing 15sec. from 2 to 2 minutes, due to `iowait` I believe.
> This is not a verdict, but, in my opinion, with the growth on DB, more I/O, mem. required, "this" will not work for much longer.
> 
> ![image](https://user-images.githubusercontent.com/83612878/134655521-c17bb870-57b7-4c61-b6c9-b3c2be6c4ecb.png)

You have to put the .chia folder in an usb ssd. the internal sd card is very slow and will die at this workload.
my .chia folder with db etc is on an nvme usb drive.

I am not harvesting on my pi node, why?
the pi has shared io. 

my Setup: 

-pi4 4gb FULL-NODE 

For every 80TB i use an new pi wich is only running a harvester and is connected to the main pi node.
